nvflare/fuel/f3/mpm.py

```diff
@@ -139,11 +140,18 @@ def run(cls, main_func, shutdown_grace_time=1.5, cleanup_grace_time=1.5):
                 f"{cls.name}: the mpm.run() method is called from {t.name}: it must be called from the MainThread"
             )
 
+        if not run_dir:
+            running_folder = os.getcwd()
```

Comment:
Could there be cases where multiple different programs that somehow their `os.getcwd()` is the same?

In those cases we will have racing issue?
Have we considered all possible scenarios?

(Multiple clients or multiple servers or multiple jobs)
(Or 2 NVFlare systems on a same machine)

Reply:
each server and client has its own workspace. We only allow one instance to run within the same workspace due to the configuration. 
BTW, fixed the variable name error.